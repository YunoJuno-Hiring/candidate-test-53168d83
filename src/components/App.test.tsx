import React from 'react';
import { createRoot } from 'react-dom/client';
import * as ShallowRenderer from 'react-test-renderer/shallow';
import { App } from './App';

describe('App Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    const root = createRoot(div!);
    root.render(<App />);
    root.unmount();
  });

  it('Matches DOM snapshot', () => {
    const renderer = ShallowRenderer.createRenderer();
    const domTree = renderer.render(<App />);
    expect(domTree).toMatchSnapshot();
  });
});
