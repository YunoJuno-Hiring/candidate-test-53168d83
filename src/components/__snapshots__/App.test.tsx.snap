// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Matches DOM snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Lord of the Rings Character Index
    </h1>
  </header>
  <section
    className="App-content"
  >
    <Layout>
      <CharacterList />
    </Layout>
  </section>
</div>
`;
